python/lsst/meas/algorithms/detection.py

```diff
@@ -489,6 +515,38 @@ def setEdgeBits(maskedImage, goodBBox, edgeBitmask):
                                                      afwGeom.ExtentI(w, h)), afwImage.LOCAL)
             edgeMask |= edgeBitmask
 
+    @pipeBase.timeMethod
+    def replacePeaks(self, targetFpList, goodPeakFpList):
+        """Replace all of the peaks in one FootprintSet with the overlapping ones from another.
+
+        When this would remove all of the Peaks from a Footprint, we instead remove all but the
+        brightest.
+
+        Parameters
+        ----------
+        targetFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to replace.
+        goodPeakFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to use instead.
+        """
+        for targetFp in targetFpList:
+            targetBBox = targetFp.getBBox()
+            newPeaks = afwDet.PeakCatalog(targetFp.getPeaks().getTable())
+            for goodPeakFp in goodPeakFpList:
```

Comment:
I don't think I can remove things from `goodPeakFpList`; not only is deleting in a loop tricky, it's possible one of those footprints could overlap multiple footprints in the other list.

But sorting beforehand (maybe by ymax?) is a good idea.  I'll have a go at implementing that.